add pipeline stage id (#326)
Oneflow-Inc/libai: docs/source/tutorials/basics/Config_System.md

        data_parallel_size=1,
        tensor_parallel_size=1,
        pipeline_parallel_size=1,
        # set pipeline_num_layers when pipeline_parallel_size > 1
        pipeline_num_layers=None,
        # custom_pipeline_stage_id could be set for adjust num_layers in different stages
        # it is usually used for manually balance calculation between stages in pipeline_parallelism
        # e.g. you can set `custom_pipeline_stage_id=[0, 0, 0, 1]` 
        # for `pipeline_num_layers=4 and pipeline_parallel_size=2`
        # it means the first 3 layers will be located in stage0 and 
        # the last layer will located in stage1
        # NOTE: if it is None, LiBai will automatically set pipeline_stage_id
        # `auto_pipeline_stage_id` and `actual_pipeline_stage_id` will be saved in `config.yaml`
        custom_pipeline_stage_id=None,
    ),

>     # device type of input tensor in model, in most cases set it to "cuda".

Reviewer: ```suggestion
    # the device type of input tensors for model, defaults to "cuda".
```

>     # when pipeline_parallel > 1 and you want to ultimate acceleration in model training

Reviewer: ```suggestion
    #if you want to accelerate the model training when pipeline_parallel > 1
```

>     # set `input_placement_device="cpu"` and use tensor.to_global() in your model.forward()

Reviewer: ```suggestion
    # you can set `input_placement_device="cpu"` then call input_tensor.to_global() inside your model.forward() method
```

    # see `libai/models/bert_model.py` as reference
    input_placement_device="cuda",

>     # rdma enabled, set it to `True` for improving speed of pipeline_parallel

Reviewer: ```suggestion
    # set to `True` to enable rdma for improving speed of pipeline_parallel
```
